articles/quickstart/backend/php/01-authorization.md

```diff
@@ -1,251 +1,184 @@
+require 'vendor/autoload.php';
+
+// Load our environment variables from the .env file:
+(Dotenv\Dotenv::createImmutable(__DIR__))->load();
+
+// Create a configuration object for the Auth0 PHP SDK:
+$auth0Configuration = new \Auth0\SDK\SdkConfiguration(
+    domain: $env['AUTH0_DOMAIN'],
+    clientId: $env['AUTH0_CLIENT_ID'],
+    clientSecret: $env['AUTH0_CLIENT_SECRET'],
+    audience: $env['AUTH0_AUDIENCE']
+);
+
+// Now instantiate the Auth0 class with the above configuration:
+$auth0 = new \Auth0\SDK\Auth0($auth0Configuration);
 ```
 
-## Protect API Endpoints
 
-<%= include('../_includes/_api_endpoints') %>
+### Authenticating the user
 
-The `before` hook from the **router** package can be used to configure which routes are to be protected. For example, you may wish to protect all routes under a URL of `/api/private`.
+For this sample application, we're focusing on [authorization](https://auth0.com/intro-to-iam/authentication-vs-authorization/). There's numerous routes you could go for authenticating your users before they hit your backend API for authorization, such as using [Auth0's SPA.js library](https://github.com/auth0/auth0-spa-js). This approach is demonstrated in [this Quickstart app accompanying Github project](https://github.com/auth0-samples/auth0-php-api-samples/). Regardless of the approach you take, this sample application expects you to pass your Access Token to it through a request parameter or header to work.
 
-```php
-// index.php
+### Authorizing a token
+
+First, we need to extract the token from the incoming HTTP request. Let's look for a `?token` parameter in a GET request or an `HTTP_AUTHORIZATION` or `Authorization` header.
 
-// ...
-// Require composer autoloader
-require __DIR__ . '/vendor/autoload.php';
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
+
+$authorized = false;
+$token = $_GET['token'] ?? $_SERVER['HTTP_AUTHORIZATION'] ?? $_SERVER['Authorization'] ?? null;
+```
 
-$app = new \App\Main( 'https://${account.namespace}/', '${apiIdentifier}' );
+Next, let's decode the token, if one is present:
 
-// Create Router instance
-$router = new \Bramus\Router\Router();
+```PHP
+// 👆 We're continuing from the steps above. Append this to your index.php file.
 
-// Activate CORS
-function sendCorsHeaders() {
-  header("Access-Control-Allow-Origin: *");
-  header("Access-Control-Allow-Headers: Authorization");
-  header("Access-Control-Allow-Methods: GET,HEAD,PUT,PATCH,POST,DELETE");
+// If a token is present, process it.
+if ($token !== null) {
+    // Trim whitespace from token string.
+    $token = trim($token);
+
+    // Remove the 'Bearer ' prefix, if present, in the event we're getting an Authorization header that's using it.
+    if (substr($token, 0, 7) === 'Bearer ') {
+        $token = substr($token, 7);
+    }
+
+    // Attempt to decode the token:
+    try {
+        $token = $auth0->decode($token);
```

Comment:
is this function **verifying** the contents of the token? if so, please add a comment to the code clarifying that.

e.g. if this is an Access Token, I'd expect to check:
- token issuer (Auth0 server)
- token audience (this API)
- token expiration (still valid)
- token subject (must not be empty, as it will represent the user)
- token signature (expected algorithm must be configured by the dev)

Reply:
Good call, I've expanded the comment